### PR TITLE
Rewrite bot in TS

### PR DESCRIPTION
**Pull request by TechGeekGamer**

### CODE REVIEW

**Comment by hax4dazy** [comment]
I am hax4dayz and I approve of this pull request

**Review by hax4dazy — APPROVED** [review]

**Review by Slluxx — COMMENTED** [review]
see included comments

File: src/commands/support.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+export const supportCommands:Array<Command> = [
+    {
+        name: 'exfat',
+        description: 'Displays info on why not to use exfat',
+        execute(message, args) {
+            const exampleEmbed = new MessageEmbed()
+            .setTitle("Guiformat")
+            .setURL('http://ridgecrop.co.uk/guiformat.exe')
+            .setDescription('A useful tool for formatting SD cards over 32GB as FAT32 on Windows.')
+            message.channel.send("The exFAT drivers built into the Switch has been known to corrupt SD cards and homebrew only makes this worse. Backup everything on your SD card as soon as possible and format it to FAT32. On Windows, if your SD card is over 32GB then it will not let you select FAT32 from the built-in format tool, however you can use a tool like GUIFormat to format it.", {embed:exampleEmbed})
```
Comment (hax4dazy) [inline src/commands/support.ts:13]: Can we change this text to the same text that behemoth has. We'd have to reword it a bit tho to be more user friendly

Comment (Slluxx) [reply src/commands/support.ts:13]: how about a suggestion @hax4dazy :harold:

Comment (hax4dazy) [reply src/commands/support.ts:13]: Oh I rewrote the text and DM'ed it to him. Let me put it here tho.

`The standard exFAT driver is bad and shouldn't be used.

If your PC doesn't allow you to format your card to FAT32 you can do that in hekate
Tools -> Arch bit · RCM · Touch · Partition -> Partition`

Comment (Slluxx) [reply src/commands/support.ts:13]: how about "should be avoided at all cost" ?
and isnt "arch bit" actually "archive bit" ? idk though, never used it.

Comment (hax4dazy) [reply src/commands/support.ts:13]: Nope its actually called arch bit. Also yeah "should be avoided at all cost" does sounds better.

File: src/commands/support.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
+import {Command} from '../../typings'
+import { MessageEmbed } from 'discord.js'
+import {config} from '../../config'
+export const supportCommands:Array<Command> = [
+    {
```
[the remark this answers is not in the document]
Comment (TechGeekGamer) [reply src/commands/support.ts:5]: Added in e964cde69e793807ca3e3c1b6bb7b83a4c3a3fe2

File: src/commands/support.ts (hunk 1 above)
Comment (TechGeekGamer) [reply src/commands/support.ts]: > Can we change this text to the same text that behemoth has. We'd have to reword it a bit tho to be more user friendly

Updated in e964cde69e793807ca3e3c1b6bb7b83a4c3a3fe2

**Review by hax4dazy — CHANGES_REQUESTED** [review]
I changed the text a bit in [this](https://github.com/Team-Neptune/Korral-JS/pull/11#discussion_r635475302) discussion

**Comment by TechGeekGamer** [comment]
> I changed the text a bit in [this](https://github.com/Team-Neptune/Korral-JS/pull/11#discussion_r635475302) discussion

b50e92347b0f906241fb863031920b984a654054

File: src/commands/moderation.ts [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+                return message.channel.send(`You can't perform that action on this user.`);
+    
+            if (!warnings[mentionedUser.id])
+                warnings[mentionedUser.id] = [];
+    
+            warnings[mentionedUser.id].push(reason);
+    
+            writeFileSync('./warnings.json', JSON.stringify(warnings))
+            var eventMessage = `You were warned on ${message.guild.name}.\nThe given reason is: ${reason}\n\nPlease read the rules. This is warning #${(warnings[mentionedUser.id].length)}.`
+            switch (warnings[mentionedUser.id].length) {
```
[the remark this answers is not in the document]
Comment (TechGeekGamer) [reply src/commands/moderation.ts:224]: How would the `message` be used?

**Comment by hax4dazy** [comment]
> 
> 
> > I changed the text a bit in [this](https://github.com/Team-Neptune/Korral-JS/pull/11#discussion_r635475302) discussion
> 
> [b50e923](https://github.com/Team-Neptune/Korral-JS/commit/b50e92347b0f906241fb863031920b984a654054)

You still missed some things. Please copy the message and paste it in there

File: src/commands/moderation.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Slluxx) [reply src/commands/moderation.ts:224]: its the `eventMessage` text

[the remark this answers is not in the document]
Comment (TechGeekGamer) [reply src/commands/moderation.ts]: Alright how does the setup in d84f5ebae3dce295972cc016a36b182e7dc9e5c3 look?

[the remark this answers is not in the document]
Comment (Slluxx) [reply src/commands/moderation.ts]: looks good to me

**Review by Slluxx — APPROVED** [review]
merge and see what breaks i guess

**Comment by TechGeekGamer** [comment]
> > > I changed the text a bit in [this](https://github.com/Team-Neptune/Korral-JS/pull/11#discussion_r635475302) discussion
> > 
> > 
> > [b50e923](https://github.com/Team-Neptune/Korral-JS/commit/b50e92347b0f906241fb863031920b984a654054)
> 
> You still missed some things. Please copy the message and paste it in there

Updated it in e1abe1cba7772326d31c32d4d9a8d6ef7a098bd6, how does it look?

**Review by hax4dazy — APPROVED** [review]

**Pull request merged**